docs/sdk/v3/guides/03-web3-development-basics.md

```diff
@@ -0,0 +1,157 @@
+
+Throughout the docs you will see how to interact with the Uniswap subgraph to fetch tick data and more without ever touching RPCs.
+
+An important note about Indexers though:
+
+While they can be helpful, you need to be aware that the ultimate source of truth comes from the RPCs. As reorgs happen and
+certain issues on indexers arise, there might be certain datapoints that are either not fully up-to-date or even
+completely wrong on Indexers. If you have a use-case that requires perfectly correct data all the time, use RPCs directly.
+If you are just doing data visualization or dashboards, use Indexers if they fit your use-case.
+
+Some popular subgraphs that you can try to fetch data from the blockchain easily (click on playground to give it a try):
+
+* [Uniswap Messari subgraph](https://thegraph.com/explorer/subgraphs/ELUcwgpm14LKPLrBRuVvPvNKHQ9HvwmtKgKSH6123cr7?view=Overview&chain=mainnet)
+* [Snapshot Subgraph](https://thegraph.com/explorer/subgraphs/3Q4vnuSqemXnSNHoiLD7wdBbGCXszUYnUbTz191kDMNn?view=Overview&chain=mainnet)
+* [Aave V3 Messari subgraph](https://thegraph.com/explorer/subgraphs/HB1Z2EAw4rtPRYVb2Nz8QGFLHCpym6ByBX6vbCViuE9F?view=Overview&chain=mainnet)
+
+## Smart Contract Development
+
+Smart contracts are typically developed using the [Solidity language](https://soliditylang.org/).
+
+There are VSCode plugins that make the development with Solidity easier. One of them is Juan Blanco's "Solidity" that you can find
+over [here](https://marketplace.visualstudio.com/items?itemName=JuanBlanco.solidity).
+
+You would typically also go for a developer suite like one of the following:
+
+* [Foundry](https://github.com/foundry-rs/foundry) - Fast and modern developer tools for smart contract engineers.
+* [Hardhat](https://github.com/NomicFoundation/hardhat) - Scriptable, easy and battle-tested JS developer suite for smart contracts.
+* (deprecated) Ganache - One of the original developer suites. Should be avoided as there are many issues with it.
+
+To read data from smart contracts, or interact with them, use the [client SDKs](#client-implementations) linked to above.
+They help you generate the necessary RPC calls to fetch data from / send transactions to the blockchain to interact with your
+smart contracts. You can read more about it in the [ethers.js docs](https://docs.ethers.org/v5/api/contract/contract/).
+
+## The Uniswap SDK
+
+How does the Uniswap SDK fit into the stack explained above?
+
+The Uniswap SDK is a tool that makes it easy to get quotes for swaps, simulate swaps and actually send transactions that execute a swap
+on-chain.
+
+As you know already, all interactions with the blockchain happens through RPCs. So the Uniswap SDK, as you will see throughput the guides,
+requires you to have access to an RPC endpoint like [Chainnodes](https://www.chainnodes.org).    
+When reading data, the data is read from the given RPC endpoint. When actually swapping, you will need to sign a transaction
```

Comment:
The wording here is a bit confusing. Reading data = RPC endpoint, swapping = sign a transaction so these look like opposites. Neither is an RPC endpoint the opposite to signing a transaction nor is reading data and swapping the only interactions we have with the Uniswap protocol.
What I imagined here for these sections is a more thorough explanation of read and write operations on blockchains. This can easily be a 50-100 line section with several subpoints explaining what happens when we read or write to a blockchain.
For example, explain the difference between a blockchain and a normal database and why a write operation to a state machine does not give us a response like a POST request to a database exposed by an HTTP server.